src/utils/catalog.lisp

```diff
@@ -48,11 +48,16 @@
            table-list view-list matview-list extension-list sqltype-list)
 
 (defstruct table source-name name schema oid comment
-           storage-parameter-list tablespace
+           storage-parameter-list tablespace row-count-estimate
```

Comment:
Undefined behaviour.

In `optimize-table-copy-ordering` in `migrate-database.lisp` we are accessing the value of slot `row-count-estimate` in structure `table` without setting it first.

[CLHS DEFSTRUCT](http://clhs.lisp.se/Body/m_defstr.htm) states:
> If no `slot-initform` is supplied, the consequences are undefined if an attempt is later made to read the slot's value before a value is explicitly assigned.

We are lucky that SBCL and CCL initialize these values to `NIL`, but they are not required to.

The slot definition `row-count-estimate` should be changed to `(row-count-estimate nil)` to explicitly initialize the slot if we want to read it without setting. Orrrr, we can even initialize it to `0` if that is the value that we are looking for in `optimize-table-copy-ordering`; that will simplify the `FLET` found there into:
```lisp
           (flet ((compare-row-count-estimates (a b)
                    (declare (fixnum a b))
                    (> a b)))
```
At this point, the function becomes unnecessary, since it becomes just a `#'>` call.

Reply:
It's all simplified down thanks to better type definition, thanks!